packages/gatsby-image/src/index.js

```diff
@@ -206,22 +207,25 @@ function generateBase64Sources(imageVariants) {
 }
 
 function generateNoscriptSource({ srcSet, srcSetWebp, media, sizes }, isWebp) {
-  const src = isWebp ? srcSetWebp : srcSet
-  const mediaAttr = media ? `media="${media}" ` : ``
-  const typeAttr = isWebp ? `type='image/webp' ` : ``
-  const sizesAttr = sizes ? `sizes="${sizes}" ` : ``
+  const propsObj = {};
 
-  return `<source ${typeAttr}${mediaAttr}srcset="${src}" ${sizesAttr}/>`
+  propsObj["media"] = media;
+  if(isWebp) propsObj["type"] ='image/webp'
+  propsObj["srcset"] = isWebp ? srcSetWebp : srcSet
+  propsObj["sizes"] = sizes
+
+  return <source {...propsObj}/>
```

Comment:
Is there a special reason why the property names are dynamic strings rather than regular properties? And since you're creating the object, anyways, you may as well use the shorthand notation.

Also, it looks like prettier/linting wasn't properly applied since this codebase uses backticks for most strings. But linting should would fail in that case anyways so you'd know :)

```suggestion
  const propsObj = {
    media,
    type: isWebp ? `image/webp` : undefined,
    srcset: isWebp ? srcSetWebp : srcSet,
    sizes,
  };
```

Also, if it's relevant whether the `type` property exists, that would still have to be set conditionally afterwards as JS doesn't have conditional property setting syntax for object literals.